frontend/class-breadcrumbs.php

```diff
@@ -367,13 +367,14 @@ private function set_crumbs() {
 			}
 		}
 		elseif ( is_post_type_archive() ) {
-			$post_type = $wp_query->get( 'post_type' );
+			$shop_id = $this->woocommerce_shop_page->is_shop_page() ? $this->woocommerce_shop_page->get_shop_page_id() : ( -1 );
 
-			if ( WPSEO_Utils::is_woocommerce_active() && is_shop() ) {
-				$id = wc_get_page_id( 'shop' );
-				$this->add_single_post_crumb( $id );
+			if ( $shop_id !== -1 ) {
```

Comment:
I would choose for something like this, because the woocommerce shop page class is caching the ID :)
```php
if ( $this->woocommerce_shop_page->is_shop_page() && $this->woocommerce_shop_page->get_shop_page_id() !== -1 ) {
    $this->add_single_post_crumb( $this->woocommerce_shop_page->get_shop_page_id() );
}
```

Reply:
:+1:  It makes code more readable with minimal affect related to performance.